### PR TITLE
Initializes CarPlay

### PR DESCRIPTION
**Pull request by Archdoog**
- Introduces a very basic car play architecture (just the map view so far).
- Refactors DemoApp to support car play.

There are many rough edges, small bugs, etc. But it's a foundation we can behind to build upon rather than trying to make one mega PR. Especially since it's opt-in with setup required.

https://vimeo.com/1031779816/4242642cd3?share=copy

### CODE REVIEW

**Comment by Archdoog** [comment]
Cc @engali94

File: apple/Sources/FerrostarCarPlayUI/CarPlayNavigationView.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        navigationCamera: MapViewCamera = .automotiveNavigation(),
+        minimumSafeAreaInsets: EdgeInsets = EdgeInsets(top: 16, leading: 16, bottom: 16, trailing: 16),
+        @MapViewContentBuilder makeMapContent: () -> [StyleLayerDefinition] = { [] }
+    ) {
+        self.styleURL = styleURL
+        self.minimumSafeAreaInsets = minimumSafeAreaInsets
+
+        userLayers = makeMapContent()
+        self.navigationCamera = navigationCamera
+        // TODO: Correct me
```
Comment (ianthetechie) [inline apple/Sources/FerrostarCarPlayUI/CarPlayNavigationView.swift:54]: Not a merge blocker, but what's this comment mean?

File: apple/Sources/FerrostarCarPlayUI/Extensions/UIHostingControllerInit.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import SwiftUI
+import UIKit
+
+// swiftformat:disable all
+extension UIHostingController where Content: View {
+    /// Create a SwiftUI view in a UIHosting controller with a closure init.
+    ///
+    /// - Parameter content: The convent view builder.
+    convenience init<V: View>(@ViewBuilder content: () -> V) where V == Content {
```
Comment (ianthetechie) [inline apple/Sources/FerrostarCarPlayUI/Extensions/UIHostingControllerInit.swift:9]: Is it easy to fix the warning here? If not, we can just note it in #164.

**Review by ianthetechie — COMMENTED** [review]
Looks mergeable to me! How much effort would it be to at least add skeletal docs to the guide?

**Review by ianthetechie — APPROVED** [review]

**Pull request merged**